jconstraints-z3/src/main/java/gov/nasa/jpf/constraints/solvers/nativez3/NativeZ3SolverContext.java

```diff
@@ -391,6 +403,23 @@ public void add(final List<Expression<Boolean>> expressions) {
     }
   }
 
+  private void normalAdd(final Expr[] exprs) {
+    solver.add(exprs);
+  }
+
+  private void addAndTrack(final Expr[] exprs, final List<Expression<Boolean>> jExprs) {
+    BoolExpr[] trackerVars = new BoolExpr[exprs.length];
+    int idx = 0;
+    for (Expr e : exprs) {
+      Variable trackVar = Variable.create(BuiltinTypes.BOOL, "track" + ++tracker);
+      BoolExpr z3Var = (BoolExpr) generatorStack.peek().createBoolVar(trackVar);
+      trackerVars[idx] = z3Var;
+      tracking.put(z3Var, jExprs.get(idx));
+      ++idx;
+    }
```

Comment:
This should be equivalent to
```suggestion
    for (int idx = 0; idx < exprs.length; ++idx) {
      Variable<Boolean> trackVar = Variable.create(BuiltinTypes.BOOL, "track" + ++tracker);
      BoolExpr z3Var = (BoolExpr) generatorStack.peek().createBoolVar(trackVar);
      trackerVars[idx] = z3Var;
      tracking.put(z3Var, jExprs.get(idx));
    }
```
Also, `tracker` starts at `1` due to prefix `++`; is this intentional?